### PR TITLE
Use better random generation in benchmark and stress tests

### PR DESCRIPTION
**Pull request by cijothomas**
As shown [here](https://github.com/open-telemetry/opentelemetry-rust/pull/1667), a good amount of time in benchmarks are spend in just generating random numbers. This PR reduces the cost of that to be more manageable, and not significantly polluting the actual cost of our code!
In benchmarks, the random generator cost fell from **377 ns to 50 ns**. The stress test for random moved from **53M/sec to 950M/sec!** 

Despite this, metrics is still stuck at 3M/sec due to heavy contention, which, I'll be addressing very shortly.

Traces stress test is stuck at 9M/sec as it did not get a key [perf fix from Logs](https://github.com/open-telemetry/opentelemetry-rust/pull/1455)!

Logs are hitting 27 M/sec, but there are few in-flight PRs and other ideas @lalitb is working to make this a lot more better.

Also added current numbers in the benchmark/stress test files for easy comparison.

### CODE REVIEW

File: opentelemetry-sdk/benches/metric_counter.rs [hunk 1]
```diff
@@ -50,12 +72,11 @@ fn counter_add(c: &mut Criterion) {
...
-            let mut rng = SmallRng::from_entropy();
             // 4*4*10*10 = 1600 time series.
-            let index_first_attribute = rng.gen_range(0..4);
-            let index_second_attribute = rng.gen_range(0..4);
-            let index_third_attribute = rng.gen_range(0..10);
-            let index_forth_attribute = rng.gen_range(0..10);
+            let index_first_attribute = CURRENT_RNG.with(|rng| rng.borrow_mut().gen_range(0..4));
+            let index_second_attribute = CURRENT_RNG.with(|rng| rng.borrow_mut().gen_range(0..4));
+            let index_third_attribute = CURRENT_RNG.with(|rng| rng.borrow_mut().gen_range(0..10));
+            let index_forth_attribute = CURRENT_RNG.with(|rng| rng.borrow_mut().gen_range(0..10));
```
Comment (lalitb) [inline opentelemetry-sdk/benches/metric_counter.rs:79]: how about borrowing once, and generate all numbers in single closure:
```rust
    CURRENT_RNG.with(|rng| {
        let mut rng = rng.borrow_mut();
        let index_first_attribute = rng.gen_range(0..4);
        let index_second_attribute = rng.gen_range(0..4);
        let index_third_attribute = rng.gen_range(0..10);
        let index_forth_attribute = rng.gen_range(0..10);
        counter.add(...)
    });

```

**Review by lalitb — APPROVED** [review]
Thanks nicely done.

File: opentelemetry-sdk/benches/metric_counter.rs [hunk 2]
```diff
@@ -30,12 +53,11 @@ fn counter_add(c: &mut Criterion) {
     let counter = create_counter();
     c.bench_function("Counter_Add_Sorted", |b| {
         b.iter(|| {
-            let mut rng = SmallRng::from_entropy();
             // 4*4*10*10 = 1600 time series.
-            let index_first_attribute = rng.gen_range(0..4);
-            let index_second_attribute = rng.gen_range(0..4);
-            let index_third_attribute = rng.gen_range(0..10);
-            let index_forth_attribute = rng.gen_range(0..10);
+            let index_first_attribute = CURRENT_RNG.with(|rng| rng.borrow_mut().gen_range(0..4));
```
Comment (TommyCpp) [inline opentelemetry-sdk/benches/metric_counter.rs:57]: Can we generate four arrays filled with random numbers before we start? In each iteration, we assign an iteration index, then iterate over the pre-generated random numbers. This way, we separate the random number generation from the measurement.

**Review by utpilla — APPROVED** [review]

File: opentelemetry-sdk/benches/metric_counter.rs (hunk 2 above)
Comment (cijothomas) [reply opentelemetry-sdk/benches/metric_counter.rs]: Good idea! I'll explore that next. At this point, we got the random to be manageable, so merging this PR to unblock progress.

Comment (cijothomas) [reply opentelemetry-sdk/benches/metric_counter.rs]: https://github.com/open-telemetry/opentelemetry-rust/issues/1805

File: opentelemetry-sdk/benches/metric_counter.rs (hunk 1 above)
Comment (cijothomas) [reply opentelemetry-sdk/benches/metric_counter.rs]: thanks! It boosted throughput further!

File: opentelemetry-sdk/benches/metric_counter.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cijothomas) [reply opentelemetry-sdk/benches/metric_counter.rs]: Did that, along with borrow once and generate all randomns in one go!

**Pull request merged**